gms/gossiper.cc

```diff
@@ -2389,6 +2389,35 @@ bool gossiper::is_safe_for_bootstrap(inet_address endpoint) {
     return allowed;
 }
 
+bool gossiper::is_safe_for_restart(inet_address endpoint, utils::UUID host_id) {
```

Comment:
See 49c00fa989cb9fa107f8b114325b8df45020b7f9 making uuids type-safe so they can't be confused with each other.